A8question1.go

```diff
@@ -0,0 +1,41 @@
+//Sharanya Datrange
+
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+var mutex sync.Mutex
+
+func isEven(n int) bool {
+	return n%2 == 0
+}
+
+func main() {
+	n := 0
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		mutex.Lock()
+		n++
+		mutex.Unlock()
+		defer wg.Done()
```

Comment:
If you are explicitly and knowingly calling it at end of function then no need of defer. This should be 1st statement inside goroutine code block, so even if your code returns from somewhere middle this will ensure decrementing of counter.